Sources/Rules/EnvironmentEntry.swift

```diff
@@ -0,0 +1,146 @@
+// Created by miguel_jimenez on 10/11/24.
+// Copyright © 2024 Airbnb Inc. All rights reserved.
+
+import Foundation
+
+public extension FormatRule {
+    /// Removes types conforming `EnvironmentKey` and replaces them with the @Entry macro
+    static let environmentEntry = FormatRule(
+        help: "Updates `EnvironmentValues` to use the @Entry macro",
+        disabledByDefault: true
+    ) { formatter in
+        let declarations = formatter.parseDeclarations()
+
+        // Find all structs conforming `EnvironmentKey`
+        let environmentKeys = Dictionary(uniqueKeysWithValues: formatter.findAllEnvironmentKeys(declarations).map { ($0.key, $0) })
+
+        // Find all `EnvironmentValues` properties
+        let environmentValuesPropertiesDeclarations = formatter.findAllEnvironmentValuesProperties(declarations, referencing: environmentKeys)
+
+        // Modify `EnvironmentValues` properties by removing its body and adding the @Entry macro
+        formatter.modifyEnvironmentValuesProperties(environmentValuesPropertiesDeclarations)
+
+        // Remove `EnvironmentKey`s
+        let updatedEnvironmentKeys = Set(environmentValuesPropertiesDeclarations.map(\.key))
+        formatter.removeEnvironmentKeys(updatedEnvironmentKeys)
+    } examples: {
+        """
+        ```diff
+        - struct ScreenNameEnvironmentKey: EnvironmentKey {
+        -   static var defaultValue: Identifier? {
+        -      .init("undefined") 
+        -     }
+        -   }
+
+        -  extension EnvironmentValues {
+        -    var screenName: Identifier? {
+        -      get { self[ScreenNameEnvironmentKey.self] }
+        -      set { self[ScreenNameEnvironmentKey.self] = newValue }
+        -    }
+        -  }
+
+        +  extension EnvironmentValues {
+        +    @Entry var screenName: Identifier? = .init("undefined")
+        +  }
```

Comment:
Since the outer extension declaration itself isn't modified it doesn't need to be included twice in the diff

```suggestion
           extension EnvironmentValues {
        -    var screenName: Identifier? {
        -      get { self[ScreenNameEnvironmentKey.self] }
        -      set { self[ScreenNameEnvironmentKey.self] = newValue }
        -    }
        +    @Entry var screenName: Identifier? = .init("undefined")
           }
```